src/app/components/Tasks.tsx

```diff
@@ -10,44 +16,109 @@ interface Props {
 
 const Tasks = ({ tasks, setTask }: Props) => {
   const [title, setTitle] = useState("");
+  const [taskDate, setTaskDate] = useState("");
+  const [error, setError] = useState("");
+  const [daysRemaining, setDaysRemaining] = useState<number | undefined>();
+  const [isClient, setIsClient] = useState(false);
+
   let nextId = 1;
 
+  const calculateDaysRemaining = (dueDate: string) => {
+    const today = new Date().valueOf();
+    const due = new Date(dueDate).valueOf();
+
+    // Calculate the difference in milliseconds
+    const timeDifference: number = due - today;
+
+    // Convert milliseconds to days
+    const daysRemaining = Math.ceil(timeDifference / (1000 * 3600 * 24));
+
+    setDaysRemaining(daysRemaining);
+  };
+
+  const handleDateChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const selectedDueDate = event.target.value;
+    setTaskDate(selectedDueDate);
+
+    if (selectedDueDate && isClient) {
+      calculateDaysRemaining(selectedDueDate);
```

Comment:
no need to calculate it here. you can do it inside onSubmit or calculate at the time of displaying the tasks